feat(Product): 상품 데이터 넣기 API

Ko-udon opened the pull request:
## 변경사항

기존의 로컬에 상품 이미지를 저장하는 방식을

Aws S3를 통해 서버에 저장하도록 변경

===============================================================
<gradle>
compileOnly('org.springframework.boot:spring-boot-starter-thymeleaf')
    implementation 'org.thymeleaf.extras:thymeleaf-extras-springsecurity5'
    implementation 'commons-io:commons-io:2.11.0'   
    implementation group: 'commons-fileupload', name: 'commons-fileupload', version: '1.4' 
내용 추가

관리자 페이지 추가


## 체크 리스트 (Checklist)

pull request 전에 아래 체크 리스트들을 만족하는 지 확인한 후 체크('x') 표시를 해주시기 바랍니다.

- [x]  master 브랜치가 아니라 **develop** 브랜치에 Merge하도록 pull request를 작성 중이신가요?
- [x]  모든 **테스트**가 성공했나요?


Postman를 활용해 테스트 진행(테스트 코드도 별도로 존재)
![image](https://user-images.githubusercontent.com/79897135/219943885-fbed33ba-d361-4de0-8b85-96d71b10acd7.png)

저장할 파일 
![image](https://user-images.githubusercontent.com/79897135/219943931-b4410c70-24b8-4459-93db-631c340d740d.png)


서버에 파일이 잘 저장된 모습
![image](https://user-images.githubusercontent.com/79897135/219943892-f01f69a7-fe1d-401c-b11a-9f81b7801519.png)


![image](https://user-images.githubusercontent.com/79897135/219943900-67ad9c1b-7262-473e-940f-b151061adeb5.png)


관리자 페이지
![image](https://user-images.githubusercontent.com/79897135/220043669-09a4219f-5ee2-4adb-98c3-15b575c61a26.png)

ehddn committed:
feat(Product): 상품 데이터 넣기 API

기존의 로컬에 상품 이미지를 저장하는 방식을

Aws S3를 통해 서버에 저장하도록 변경

seyoung5744 commented on src/main/java/com/zerobase/wishmarket/domain/product/service/ProductAddService.java:
```diff
@@ -0,0 +1,65 @@
+
+        ProductCategory productCategory = ProductCategory.values()[productInputForm.getCategoryCode()];
+        String category = productCategory.toString();
+
```
예외처리 Throw 로 알려주는 것도 나쁘지 않아보여요

seyoung5744 commented on src/test/java/com/zerobase/wishmarket/domain/product/service/ProductAddServiceTest.java:
```diff
@@ -0,0 +1,43 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
```
Mock으로 테스트 해보세요~

ehddn committed:
fix(Product): 상품 데이터 넣기 S3유틸 클래스 사용 및 관리자 페이지 추가

별도의 클래스로 파일을 업로드하던 로직을

S3유틸 클래스에 정의한 업로드 메소드를 사용

상품 데이터 넣기 관리자 페이지
추가
해당 페이지 시큐리티
설정

<gradle>
compileOnly('org.springframework.boot:spring-boot-starter-thymeleaf')

implementation 'org.thymeleaf.extras:thymeleaf-extras-springsecurity5'

implementation 'commons-io:commons-io:2.11.0'
    implementation
group: 'commons-fileupload', name: 'commons-fileupload', version: '1.4'
